packages/@vue/cli-plugin-pwa/index.js

```diff
@@ -2,25 +2,39 @@ module.exports = (api, options) => {
   api.chainWebpack(webpackConfig => {
     const name = api.service.pkg.name
 
+    const userOptions = options.pwa || {}
+
     // the pwa plugin hooks on to html-webpack-plugin
     // and injects icons, manifest links & other PWA related tags into <head>
     webpackConfig
       .plugin('pwa')
         .use(require('./lib/HtmlPwaPlugin'), [Object.assign({
           name
-        }, options.pwa)])
+        }, userOptions)])
 
     // generate /service-worker.js in production mode
     if (process.env.NODE_ENV === 'production') {
-      webpackConfig
-        .plugin('sw-precache')
-          .use(require('sw-precache-webpack-plugin'), [{
-            cacheId: name,
-            filename: 'service-worker.js',
-            staticFileGlobs: [`${options.outputDir}/**/*.{js,html,css}`],
-            minify: true,
-            stripPrefix: `${options.outputDir}/`
-          }])
+      // Default to GenerateSW mode, though InjectManifest also might be used.
+      const workboxPluginMode = userOptions.workboxPluginMode || 'GenerateSW'
+      const workboxWebpackModule = require('workbox-webpack-plugin')
+      if (workboxPluginMode in workboxWebpackModule) {
+        const workBoxConfig = Object.assign({
+          cacheId: name,
+          exclude: [
+            new RegExp('\.map$'),
+            new RegExp('img/icons/'),
+            new RegExp('favicon\.ico$'),
+            new RegExp('manifest\.json$')
+          ]
+        }, userOptions.workboxOptions)
+
+        webpackConfig
+          .plugin('workbox')
+          .use(workboxWebpackModule[workboxPluginMode], [workBoxConfig])
+      } else {
+        throw new Error(`${workboxPluginMode} is not a supported Workbox webpack plugin mode. ` +
```

Comment:
I wasn't sure what the philosophy was around misconfiguration. If there's something that should be done here other than throwing an `Error`, I'm happy to change.

Reply:
Since this is a built-in plugin, you can add it the user options validation schema here: https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli-service/lib/options.js#L3-L35

But throwing an error is also fine.

Reply:
I ❤️ `joi`, but have found the default error messages to be a little hard to read, so I'll just stick with a custom `Error` for this specific failure.